eng/doc/fips/UserGuide.md

```diff
@@ -523,13 +529,140 @@ The message is signed using [BCryptSignHash].
 
 </details>
 
+### [crypto/ecdh](https://pkg.go.dev/crypto/ecdh)
+
+Package ecdh implements Elliptic Curve Diffie-Hellman over NIST curves and Curve25519.
+
+**Implementation**
+
+All supported curves implement the `ecdh.Curve` interface as follows:
+
+<details><summary>OpenSSL (click for details)</summary>
+
+ - `GenerateKey` uses [EVP_PKEY_keygen].
+ - `NewPrivateKey` uses [EVP_PKEY_new].
+ - `NewPublicKey` uses [EVP_PKEY_new].
+
+</details>
+
+<details><summary>CNG (click for details)</summary>
+
+ - `GenerateKey` uses [BCryptGenerateKeyPair] and [BCryptExportKey].
+ - `NewPrivateKey` uses [BCryptImportKeyPair].
+ - `NewPublicKey` uses [BCryptImportKeyPair].
+
+</details>
+
+#### func [P256](https://pkg.go.dev/crypto/ecdh#P256)
+
+```go
+func ecdh.P256() ecdh.Curve
+```
+
+P256 returns a Curve which implements NIST P-256.
+
+**Implementation**
+
+<details><summary>OpenSSL (click for details)</summary>
+
+The curve uses `NID_X9_62_prime256v1`.
+
+</details>
+
+<details><summary>CNG (click for details)</summary>
+
+The curve uses `BCRYPT_ECC_CURVE_NISTP256`.
+
+</details>
+
+#### func [P384](https://pkg.go.dev/crypto/ecdh#P384)
+
+```go
+func ecdh.P384() ecdh.Curve
+```
+
+P384 returns a Curve which implements NIST P-384.
+
+**Implementation**
+
+<details><summary>OpenSSL (click for details)</summary>
+
+The curve uses `NID_secp384r1`.
+
+</details>
+
+<details><summary>CNG (click for details)</summary>
+
+The curve uses `BCRYPT_ECC_CURVE_NISTP384`.
+
+</details>
+
+#### func [P521](https://pkg.go.dev/crypto/ecdh#P521)
+
+```go
+func ecdh.P521() ecdh.Curve
+```
+
+P521 returns a Curve which implements NIST P-521.
+
+**Implementation**
+
+<details><summary>OpenSSL (click for details)</summary>
+
+The curve uses `NID_secp521r1`.
+
+</details>
+
+<details><summary>CNG (click for details)</summary>
+
+The curve uses `BCRYPT_ECC_CURVE_NISTP521`.
+
+</details>
+
+#### func [X25519](https://pkg.go.dev/crypto/ecdh#X25519)
+
+ecdh.X25519 is not implemented by any backend.
+
+#### func [PrivateKey.ECDH](https://pkg.go.dev/crypto/ecdh#PrivateKey.ECDH)
+
+```go
+func (k *ecdh.PrivateKey) ECDH(remote *ecdh.PublicKey) ([]byte, error)
+```
+
+ECDH performs a ECDH exchange and returns the shared secret. The PrivateKey and PublicKey must use the same curve.
+
+**Requirements**
+
+- `remote` must be an object created from `ecdh.P256()`, `ecdh.P384()`, or `ecdh.P521()`.
+
+**Implementation**
+
+<details><summary>OpenSSL (click for details)</summary>
+
+The key is derived using [EVP_PKEY_derive].
+
+</details>
+
+<details><summary>CNG (click for details)</summary>
+
+The key is derived using [BCryptDeriveKey].
+
+</details>
+
 ### [crypto/ed25519](https://pkg.go.dev/crypto/ed25519)
 
 Not implemented by any backend.
 
 ### [crypto/elliptic](https://pkg.go.dev/crypto/elliptic)
 
-Not implemented by any backend.
+Not implemented by any backend, but the following `elliptic.Curve` constructors can be used together with `crypto/ecdsa`.

```

Comment:
I think this would be easier to understand if it were more specific... here's a shot at it. (Otherwise it seems to me like a contradiction to say it's not implemented by a backend yet it's ok to do.)

```suggestion
Not implemented by any backend, but to use `ecdsa.GenerateKey`, one of the following `elliptic.Curve` constructors must be used to specify the curve. See [`ecdsa.GenerateKey`](#func-generatekey) for additional requirements. As long as the requirements are met, only the name of the curve is used, not the curve parameters or methods implemented by standard Go crypto, allowing FIPS compliance.
```